aten/src/ATen/core/DistributionsHelper.h

```diff
@@ -117,28 +117,36 @@ struct normal_distribution {
 
   inline dist_acctype<T> operator()(at::CPUGenerator* generator){
     dist_acctype<T> ret;
-    bool is_cache_available = generator->is_normal_cache_available();
-    if (!is_cache_available) {
-      uniform_real_distribution<T> uniform(0.0, 1.0);
-      const dist_acctype<T> u1 = uniform(generator);
-      const dist_acctype<T> u2 = uniform(generator);
-      const dist_acctype<T> r = ::sqrt(static_cast<T>(-2.0) * ::log(static_cast<T>(1.0)-u2));
-      const dist_acctype<T> theta = static_cast<T>(2.0) * static_cast<T>(M_PI) * u1;
-      if (std::is_same<T, double>::value) {
-        dist_acctype<double> cache = r * ::sin(theta) * stdv + mean;
-        generator->set_normal_cache_double(cache);
-      } else {
-        dist_acctype<float> cache = r * ::sin(theta) * stdv + mean;
-        generator->set_normal_cache_float(cache);
+    // return cached values if available
```

Comment:
Non-actionable comment: you are so lucky float and double are interconvertible, otherwise this wouldn't have typechecked :3

Reply:
Oops :)